server-backend/src/LiquidService.ts

```diff
@@ -0,0 +1,112 @@
+import axios from 'axios';
+
+
+export class LiquidConfigurationDetails {
+    readonly rpcUrl!: string;
+    readonly rpcAuth!: {username: string, password: string};
+    public xpub!: string;
+}
+
+@Injectable({ scope: Scope.DEFAULT })
+export class LiquidService implements OnApplicationBootstrap  {
+
+    configurationDetails: LiquidConfigurationDetails;
+    public readonly xpub: string;
+    private readonly logger = new Logger(LiquidService.name);
+    private readonly rpcUrl: string;
+    private readonly rpcAuth: {username: string, password: string};
+    
+    constructor(
+        private nbxplorer: NbxplorerService,
+        @Inject('ELEMENTS_CONFIG') private elementsConfig: FourtySwapConfiguration['elements'],
+    ) {
+        this.xpub = this.elementsConfig.xpub;
+        this.rpcUrl = this.elementsConfig.rpcUrl;
+        this.rpcAuth = {
+            username: this.elementsConfig.rpcUsername,
+            password: this.elementsConfig.rpcPassword,
+        };
+        this.configurationDetails = {
+            rpcUrl: this.rpcUrl,
+            rpcAuth: this.rpcAuth,
+            xpub: this.xpub,
+        };
+    }
+    
+    async onApplicationBootstrap(): Promise<void> {
+        this.logger.debug('Starting to initialize LiquidService xpub');
+        try {
+            await this.nbxplorer.track(this.xpub, 'lbtc');
+            this.logger.debug('LiquidService xpub initialized successfully');
+        } catch (error: unknown) {
+            const errorMessage = error instanceof Error ? error.message : 'Unknown error';
+            this.logger.error(`Failed to initialize Liquid xpub: ${errorMessage}`);
+        }
+    }
+
+    async callRPC(method: string, params: any[] = [], walletName: string = ''): Promise<any> {
+        const url = walletName ? `${this.rpcUrl}/wallet/${walletName}` : this.rpcUrl;
+        try {
+            const response = await axios.post(url, {
+                jsonrpc: '1.0',
+                id: '40swap',
+                method,
+                params,
+            }, {
+                auth: this.rpcAuth,
+            });
+            return response.data.result;
+        } catch (error) {
+            this.logger.error(`Error calling Elements RPC ${method}: ${(error as any).message}`);
+            throw error;
+        }
+    }
+
+    async getUtxos(): Promise<NBXplorerUtxosResponse | void> {
+        if (!this.xpub) {
+            this.logger.error('Attempting to get UTXOs with empty xpub');
+            throw new Error('Xpub is not initialized');
+        }
+        const utxoResponse = await this.nbxplorer.getUTXOs(this.xpub, 'lbtc');
+        if (!utxoResponse) {
+            throw new Error('No UTXOs returned from NBXplorer');
+        }
+        return utxoResponse;
+    }
+
+    async getConfirmedUtxos(): Promise<NBXplorerUtxo[]> {
+        const utxoResponse = await this.getUtxos();
+        return utxoResponse?.confirmed.utxOs ?? [];
```

Comment:
I think we can remove all the question marks. Typescript will complain if anything can be null, which I think is not the case here.
```suggestion
        return utxoResponse.confirmed.utxOs;
```

Reply:
Removed